File: 第四章 消息推送、使用电话本、发送邮件、短信、拨打电话/实验2_创建应用发送邮件、短信、拨打电话/MZEmailer/MZEmailer/ViewController.m
//
//  ViewController.m
//  MZEmailer
//
//  Created by 祝海焜 on 15/12/21.
//  Copyright © 2015年 Matin.Zhou. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)SendEmail:(id)sender {
    
    // Email主题
    NSString *eSub = @"Test Email";
    // Email内容
    NSString *mBody = @"M9d3 EMailer app";
    // Email收件人
    NSArray *recipients = [NSArray arrayWithObject:@"john@gmail.com"];
    
    MFMailComposeViewController *Ecomp = [[MFMailComposeViewController alloc] init];
    Ecomp.mailComposeDelegate = self;
    [Ecomp setSubject:eSub];
    [Ecomp setMessageBody:mBody isHTML:YES];
    [Ecomp setToRecipients:recipients];
    // 使mail界面显示在屏幕上
    [self presentViewController:Ecomp animated:YES completion:NULL];
}

- (void) mailComposeController:(MFMailComposeViewController *)controller
           didFinishWithResult:(MFMailComposeResult)result
                         error:(NSError *)error
{
    switch (result)
    {
        case MFMailComposeResultCancelled:
            NSLog(@"Mail cancelled");
            break;
        case MFMailComposeResultSaved:
            NSLog(@"Mail saved");
            break;
        case MFMailComposeResultSent:
            NSLog(@"Mail sent");
            break;
        case MFMailComposeResultFailed:
            NSLog(@"Mail sent failure: %@", [error localizedDescription]);
            break;
        default:
            break;
    }
    
    // 关闭mail界面
    [self dismissViewControllerAnimated:YES completion:NULL];
    
}

- (IBAction)SendSMS:(id)sender {
    
    MFMessageComposeViewController *controller = [[[MFMessageComposeViewController alloc] init] self];
    if([MFMessageComposeViewController canSendText])
        {
        //设置短信内容
        controller.body = @"This is Test SMS from Emailer app.";
        //设置短信接收号码
        controller.recipients = [NSArray arrayWithObjects:@"12345678", nil];
        controller.messageComposeDelegate = self;
    
        [self presentViewController:controller animated:YES completion:nil];
    }
    }

- (void)messageComposeViewController:(MFMessageComposeViewController *)controller
                 didFinishWithResult:(MessageComposeResult)result
    {
        switch (result) {
            case MessageComposeResultCancelled:
                //测试，点击取消时打印如下内容
                NSLog(@"Cancelled");
                break;
            case MessageComposeResultSent:
                //测试，点击发送时打印如下内容
                NSLog(@"Message sent");
                break;
            default:
                break;
    }
        [self dismissViewControllerAnimated:NO completion:nil];
}

- (IBAction)DialPhone:(id)sender{
    
    [[UIApplication sharedApplication]openURL:[NSURL URLWithString:@"tel:+8612345678"]];

}



@end

File: 第二章Swift介绍与UI设计/实验1：创建一个简单的计算器程序Dorm/MZSimpleCal_1/MZSimpleCal_1/ViewController.swift
//
//  ViewController.swift
//  MZSimpleCal_1
//
//  Created by 祝海焜 on 15/12/8.
//  Copyright © 2015年 Matin.Zhou. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    var isTyping = false
    var firstnum = 0
    var secondnum = 0
    var result = 0
    var operation = ""
    
    @IBOutlet weak var OutScreen: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    
    @IBAction func calButton(sender: AnyObject) {
        let number = sender.currentTitle
        if isTyping {
            OutScreen.text = OutScreen.text! + number!!
        } else {
            OutScreen.text = number
            isTyping = true
        }
    }
   
    @IBAction func Clear(sender: AnyObject) {
        firstnum = 0
        secondnum = 0
        operation = ""
        isTyping = false
        result = 0
        OutScreen.text = "\(result)"
    }
    
    @IBAction func operation(sender: AnyObject) {
        isTyping = false
            
        //之前代码为firstnum = OutScreen.text!.toInt()!，现在Swift为2.0版本，语法变化
        firstnum = Int(OutScreen.text!)!
        
        operation = sender.currentTitle!!
        print(operation)
    }
    
    @IBAction func equal(sender: AnyObject) {
        
        //之前代码为secondnum = OutScreen.text!.toInt()!，现在Swift为2.0版本，语法变化
        secondnum = Int(OutScreen.text!)!
        
        isTyping = false
        
        if (operation == "+")
        {  result = firstnum + secondnum
            OutScreen.text = "\(result)"
            print(result)}
            
            else if (operation == "-" )
        {   result = firstnum - secondnum
            OutScreen.text = "\(result)"
            print(result) }
            
            else if (operation == "x")
        {  result = firstnum * secondnum
            OutScreen.text = "\(result)"
            print(result)}
            
            else if (operation == "/")
        {   result = firstnum / secondnum
            OutScreen.text = "\(result)"
            print(result)
        }
    }

}

